dbt/include/trino/macros/materializations/incremental.sql

```diff
@@ -59,6 +61,54 @@
     )
 {%- endmacro %}
 
+{% macro trino__get_merge_sql(target, source, unique_key, dest_columns, predicates=none) -%}
+    {%- set predicates = [] if predicates is none else [] + predicates -%}
+    {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
+    {%- set dest_cols_csv_source = dest_cols_csv.split(', ') -%}
+    {%- set update_columns = config.get('merge_update_columns', default = dest_columns | map(attribute="quoted") | list) -%}
+    {%- set sql_header = config.get('sql_header', none) -%}
+
+    {% if unique_key %}
+        {% if unique_key is sequence and unique_key is not mapping and unique_key is not string %}
+            {% for key in unique_key %}
+                {% set this_key_match %}
+                    DBT_INTERNAL_SOURCE.{{ key }} = DBT_INTERNAL_DEST.{{ key }}
+                {% endset %}
+                {% do predicates.append(this_key_match) %}
+            {% endfor %}
+        {% else %}
+            {% set unique_key_match %}
+                DBT_INTERNAL_SOURCE.{{ unique_key }} = DBT_INTERNAL_DEST.{{ unique_key }}
+            {% endset %}
+            {% do predicates.append(unique_key_match) %}
+        {% endif %}
+    {% else %}
+        {% do predicates.append('FALSE') %}
+    {% endif %}
+
+    {{ sql_header if sql_header is not none }}
+
+    merge into {{ target }} as DBT_INTERNAL_DEST
+        using {{ source }} as DBT_INTERNAL_SOURCE
+        on {{ predicates | join(' and ') }}
+
+    {% if unique_key %}
+    when matched then update set
```

Comment:
Why should we do `set` the unique key components when we know that they are equal with the source row?

Interestingly I see the same behavior in dbt-core https://github.com/dbt-labs/dbt-core/blob/1071a4681df91633301fdf23e34de819b66fbeb7/core/dbt/include/global_project/macros/materializations/models/incremental/merge.sql#L36-L39 for `merge`. 
Am I missing something ?

Reply:
Do we expect any performance improvement by omitting the unique key components?

Reply:
Being fairly inexperienced myself in this regard, I asked for feedback on Trino's slack 

https://trinodb.slack.com/archives/CP1MUNEUX/p1661228413753349

tldr; 

>  The columns you don’t update, are still passed thru (identity), so it’s the same situation.

@hovaesco please consider my request optional. In the current situation i find it rather cosmetical - an improvement for the human readers :)

Reply:
Thanks for raising the topic with the Trino community. This is basically a copy and paste from dbt-core.